packages/fxa-payments-server/server/lib/amplitude.js

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const { GROUPS, initialize } = require('../../../fxa-shared/metrics/amplitude');
+const log = require('./logging/log')();
+const ua = require('./user-agent');
+const { version: VERSION } = require('../../package.json');
+
+const FUZZY_EVENTS = new Map([
+  [
+    // Emit events from the front-end as `amplitude.${GROUP}.${EVENT}`
+    /^amplitude\.([\w-]+)\.([\w-]+)$/,
+    {
+      group: group => GROUPS[group],
+      event: (group, event) => event,
+    },
+  ],
+]);
+
+const transform = initialize({}, {}, FUZZY_EVENTS);
+
+module.exports = (event, request, data) => {
+  if (!event || !request || !data) {
+    return;
+  }
+
+  const userAgent = ua.parse(request.headers['user-agent']);
+
+  const amplitudeEvent = transform(event, {
+    version: VERSION,
+    ...mapBrowser(userAgent),
+    ...mapOs(userAgent),
+    ...mapFormFactor(userAgent),
+    ...mapLocation(data.location),
+    ...data,
+  });
+
+  if (amplitudeEvent) {
+    log.info('amplitudeEvent', amplitudeEvent);
+  }
+};
+
+function mapBrowser(userAgent) {
```

Comment:
All of the code from here down is copy/pasted from the content server so could be refactored to `fxa-shared` if you like.